x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/datafeed/DatafeedConfig.java

```diff
@@ -613,7 +612,7 @@ public Builder(DatafeedConfig config) {
             this.scriptFields = config.scriptFields == null ? null : new ArrayList<>(config.scriptFields);
             this.scrollSize = config.scrollSize;
             this.chunkingConfig = config.chunkingConfig;
-            this.headers = new HashMap<>(config.headers);
+            this.headers = new LinkedHashMap<>(config.headers);
```

Comment:
The user never sees the headers we store.  It may be best to just stick with a `HashMap`, because that's what rollups and watcher use.